tests/checkmk-docker-hooks/post-start/generate_random_rrd_data.py

```diff
@@ -0,0 +1,148 @@
+"""
+Tool to create data in rrd files.
+Paths are hard coded for usage inside of cmk docker container.
+
+* using watchdog to wait for modification event to detect recent created rrd
+  files, but after the configuration is written to the rrd files.
+* as no old data can be inserted into rrd files, the rrd files are then
+  recreated using the configuration of the original rrd files
+* then random data for the previous 5 hours is inserted
+* different random data generator (random, static, sine) are used to generate
+  the data
+* an internal list of modified files makes sure files are only updated once
+"""
+
+
+import math
+import os
+import random
+import re
+import time
+from collections import defaultdict
+
+import rrdtool
+from watchdog.events import PatternMatchingEventHandler
+from watchdog.observers import Observer
+
+
+def random_random_function():
+    choice = random.randint(0, 2)
+    if choice == 0:
+        value = random.randint(0, 500)
+        return lambda x: value
+    if choice == 1:
+        return lambda x: random.random() * 100
+    if choice == 2:
+        offset = random.random()
+        frequency = random.random()
+        return lambda x: math.sin(offset + x / (1000.0 * (1 + frequency))) * 100
+    raise Exception()
+
+
+def clone(original_rrd, clone_rrd):
+    info = rrdtool.info(original_rrd)
+
+    metrics = set()
+    for key, value in info.items():
+        match = re.match(r"^ds\[([0-9]+)\]\.([^.]+)$", key)
+        if match:
+            id, ds_key = match.groups()
+            metrics.add(id)
+            if ds_key == "type" and value != "GAUGE":
+                raise NotImplementedError(value)
+            if ds_key == "minimal_heartbeat" and value != 8460:
+                raise NotImplementedError(value)
+    metrics_count = len(metrics)
+
+    rra_info = defaultdict(dict)
+    for key, value in info.items():
+        match = re.match(r"^rra\[([0-9]+)\]\.([^.]+)$", key)
+        if match:
+            id, rra_key = match.groups()
+            rra_info[id][rra_key] = value
+
+    dss = []
+    for i in range(1, metrics_count + 1):
+        dss.append(f"DS:{i}:GAUGE:8460:U:U")  # assuming fixed value here, see above
+    rras = []
+    for id, rra_info in rra_info.items():
+        cf = rra_info["cf"]
+        xff = rra_info["xff"]
+        rows = rra_info["rows"]
+        steps = rra_info["pdp_per_row"]
+        rras.append(f"RRA:{cf}:{xff}:{steps}:{rows}")
+
+    rrdtool.create(
+        clone_rrd,
+        "--start",
+        "0",
+        "--step",
+        "60",
+        *dss,
+        *rras,
+    )
+
+    return metrics_count
+
+
+def modify_in_place(filename):
+    filename_clone = "/tmp/clone.rrd"
+    metrics_count = clone(filename, filename_clone)
+
+    random.seed(filename)
```

Comment:
That means that the every unique `.rrd` file will have the same content, won't it? If so, I think that's beautiful.

Reply:
the curves probably won't be 100% the same, but they should have at least the same method picked each time. so if its a static value, it will be a static value in the next run.

Reply:
There's one potential problem I see with that. If we want to assert that a graph has a certain shape, we could use something simple like snapshot testing if the rrd files always have the same content. We should probably talk to @MatteoStifano about it and ask him, how far he is regarding that.

Reply:
then we would have to fake the time in the grafana container, checkmk container and also in the browser cypress uses.
first i thought we could use a fixed period of time in the past, but checkmk only keeps the last 400 days of historic data, so we would have to adapt this from time to time. from my perspective it's not important what exactly the graphs show, but simply that they show something. perhaps this can be achieved by inspecting the html?
i would love to see a test making sure the aggregation is working correctly. this could be achieved by changing the output to table and do the calculations on the table values to check if the aggregation is working in the expected way.

Reply:
The table variant sounds even better than testing something about the UI elements. I like that.